### PR TITLE
Removing go mod replace for `collector/service` and `collector/otelcol`

### PR DESCRIPTION
**Pull request by ptodev**
In the spirit of "[a little copying is better than a little dependency.](https://go-proverbs.github.io/)"....

I will also make a similar change to Agent. However, in the Agent I think we will still need the "service" package to be replaced so that we can embed the OTel Collector's metrics in the Agent's `/metrics` endpoint.

### CODE REVIEW

**Comment by ptodev** [comment]
This copying is not "a little", but I think it's still better than the go mod replace 😄

File: go.mod [hunk 1]
```diff
@@ -915,18 +915,6 @@ replace (
...
 
-// Replacing for an internal fork which allows us to observe metrics produced by the Collector.
-// This is a temporary solution while a new configuration design is discussed for the collector. Related issues:
-// https://github.com/open-telemetry/opentelemetry-collector/issues/7532
-// https://github.com/open-telemetry/opentelemetry-collector/pull/7644
-// https://github.com/open-telemetry/opentelemetry-collector/pull/7696
-// https://github.com/open-telemetry/opentelemetry-collector/issues/4970
-replace (
-	go.opentelemetry.io/collector/otelcol => github.com/grafana/opentelemetry-collector/otelcol v0.0.0-20241104164848-8ea9d0a3e17a
-	go.opentelemetry.io/collector/processor/batchprocessor => github.com/grafana/opentelemetry-collector/processor/batchprocessor v0.0.0-20241104164848-8ea9d0a3e17a
```
Comment (ptodev) [inline go.mod:926]: I removed the batch processor replace too, because this change has been merged: https://github.com/grafana/alloy/pull/2027

**Review by mattdurham — APPROVED** [review]
Fantastic LGTM, I didnt look to closely at the forked code assumed it was the same.

**Comment by dehaansa** [comment]
To avoid copied code, I reimplemented the feature using the exported tools otelcol expects you to use in this other PR: https://github.com/grafana/alloy/pull/2148

**Comment by ptodev** [comment]
Closing this PR in favour of #2148, which doesn't require copying code 🥳

**Pull request closed without merging**